cpplite/cpplite.debugger/src/org/netbeans/modules/cpplite/debugger/ni/NIDebuggerProviderImpl.java

```diff
@@ -119,9 +121,15 @@ public void finished() {
                     completed.complete(null);
                 }
             });
+            started.release();
             debugger.execRun();
             return engineProcess.second();
         }, executionDescriptor, displayName).run();
+        // Wait for the debugger to actually start up.
+        // This is necessary to be able to safely call other methods on the NIDebuggerProvider.
+        try {
+            started.acquire();
+        } catch (InterruptedException ex) {}
```

Comment:
Nitpick: if thread's `interrupted` status happen to be true, `started` won't be acquired (waited for) but the code will continue. OK ?

Reply:
It's true that it's safer to replace it with `acquireUninterruptibly`. Done, thanks.